### PR TITLE
fix: 10/8/2014 5:08:35 PM → M/D/YYYY h:mm:ss a (#20)

### PR DESCRIPTION
**Pull request by harsimranb**
1. Added a new function appendAMPMCase(); which appends the correct case
for the time period based on the AM/pm case in the date string.
2. Added a new test to the index.html page for this
change.

### CODE REVIEW

**Comment by gr2m** [comment]
Thanks, looks good!

Could you please also add a test to `tests/moment.parseFormat-test.js`? You can run tests with `grunt test`. If you haven't yet, you need to install grunt with `npm install -g grunt-cli` and install the npm dependencies with `npm install` before.

Tests could look like this:

```js
  // https://github.com/gr2m/moment.parseFormat/issues/20
  equal( moment.parseFormat('10/8/2014 5:08:35 PM'), 'M/D/YYYY h:mm:ss A', '#20 10/8/2014 5:08:35 PM → M/D/YYYY h:mm:ss A' );
  equal( moment.parseFormat('10/8/2014 5:08:35am'), 'M/D/YYYY h:mm:ssa', '#20 10/8/2014 5:08:35 am → M/D/YYYY h:mm:ssa' );
```

File: moment.parseFormat.js [hunk 1]
```diff
@@ -95,17 +100,17 @@
     format = format.replace(regexTimezone, 'Z');
 
     // 05:30:20pm ☛ hh:mm:ssa
-    format = format.replace(regexHoursWithLeadingZeroDigitMinutesSecondsAmPm, 'hh:mm:ss$1a');
+    format = format.replace(regexHoursWithLeadingZeroDigitMinutesSecondsAmPm, appendAMPMCase(format, 'hh:mm:ss$1'));
```
Comment (gr2m) [inline moment.parseFormat.js]: I wonder if we could find a way to not run the `appendAMPMCase` 6 times. Maybe we could set `amOrPm` in lie 36 to lowercase / uppercase, depending wether `(AM|PM)` or `(am|pm)` matches?

Comment (harsimranb) [inline moment.parseFormat.js]: Good idea.  I'll make that change and the tests as well.

Comment (harsimranb) [inline moment.parseFormat.js]: Actually, can you elaborate more on that solution?  We still have to append the correct 'a' or 'A' to the format based on the AM/pm casing.  Right now its hardcoded to lowercase 'a'.  Another way we can avoid 6 calls, is by adding the 'A' character around line 114 when the time formats have run.  Thoughts?

**Comment by gr2m** [comment]
merged via bd9a73d, cheers :+1:

**Pull request closed without merging**